### PR TITLE
River_Sound OK and Documented + River_Sound Related `z_actor`/`Audio` Functions

### PR DESCRIPTION
**Pull request by engineer124**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

This matches and documents `En_River_Sound`. 

In addition, 1 function from OoT's River_Sound was moved into `z_actor` but it still used by MM's River_Sound i.e. `func_800BCCDC`, which is documented here.

Also connected to `En_River_Sound` are 12 functions from the audio file `code_8019AF00.c`. 10 of these functions are completely unused in MM and are simply copy-pasted from OoT. I am including them here due to their connection with `En_River_Sound` and to shorten the eventual `code_8019AF00.c` PR.

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]
Nice work.
Just a minor comment

File: src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c [hunk 1]
```diff
@@ -27,10 +26,68 @@ const ActorInit En_River_Sound_InitVars = {
...
+    static f32 freqScale[] = {
+        0.7f, // 1 / sqrt(2)
+        1.0f, // 1
+        1.4f, // sqrt(2)
+    };
+    EnRiverSound* this = THIS;
+    s16 params = this->actor.params;
+
+    if (params < RS_RIVER_DEFAULT_LOW_FREQ) {
+        Actor_PlaySfxAtPos(&this->actor, gAudioEnvironmentalSfx[params]);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c:89]: I just realized something about this check. `RS_RIVER_DEFAULT_LOW_FREQ` has value 0xFD, so any value lower than that could be used to access `gAudioEnvironmentalSfx`, which only has 0x16 elements.
So, this _could_ be a problem, but I doubt this actor is spawned with the wrong flags on normal conditions (I haven't checked tho).
So, I guess this should be noted as a comment. I'm not sure if it should be labelled as a bug, feels a bit overkill lol

File: src/code/audio/code_8019AF00.c [hunk 2]
```diff
@@ -2087,13 +2129,130 @@ const char sAudioOcarinaUnusedText7[] = "check is over!!! %d %d %d\n";
...
+        if (targetVolume < 0x40) {
+            // Only the first room
+            lowPassFilterCutoff = 0x10;
+        } else {
+            // Higher volume leads to a higher cut-off frequency in the low-pass filtering
+            lowPassFilterCutoff = (((targetVolume - 0x40) >> 2) + 1) << 4;
+        }
+
+        Audio_QueueSeqCmd((8 << 28) | ((u8)(SEQ_PLAYER_BGM_MAIN) << 24) | ((u8)(4) << 16) | ((u8)(15) << 8) |
+                          (u8)(lowPassFilterCutoff));
```
[the remark this answers is not in the document]
Comment (engineer124) [reply src/code/audio/code_8019AF00.c:2185]: Yup, I already fully understand these `Audio_QueueSeqCmd` and the macros for those will be introduced with the decompilation and documentation of `code_80192BE0.c`. My next PR in OoT will be addressing these commands so if  you want a sneak peak...

https://github.com/engineer124/oot/blob/495fee5be19e6e3528ab135c010cde1881128b59/include/sequence.h#L307-L347

File: src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c (hunk 1 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c:89]: I think we decided that indexing an out-of-bounds array isn't enough to be considered a bug as that's up to the user to chose the available params. That's why I included every possible entry in the `RiverSoundType` enum. But I can add a comment to that enum stating that any enum not used in that list is invalid maybe?

File: src/code/audio/code_8019AF00.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (AngheloAlf) [reply src/code/audio/code_8019AF00.c:2185]: Ohh, those are a lot of macros.
Happy to hear you fully understand this function :partying_face:

File: src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c (hunk 1 above)
Comment (AngheloAlf) [reply src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c:89]: Yeah, a comment with a small note should be enough

**Review by hensldm — APPROVED** [review]

File: src/code/audio/code_8019AF00.c [hunk 3]
```diff
@@ -2147,9 +2365,58 @@ const char sAudioOcarinaUnusedText7[] = "check is over!!! %d %d %d\n";
...
+        return;
+    }
+
+    dist = sqrtf(SQ(pos->z) + (SQ(pos->x) + SQ(pos->y)));
+
+    if (sRiverSoundBgmPos == NULL) {
+        sRiverSoundBgmPos = pos;
+        func_801A3238(SEQ_PLAYER_BGM_SUB, seqId, 0, 7, 2);
+    } else {
+        prevDist = sqrtf(SQ(sRiverSoundBgmPos->z) + SQ(sRiverSoundBgmPos->x));
```
Comment (isghj5) [inline src/code/audio/code_8019AF00.c:2388]:
```suggestion
        prevDist = sqrtf(SQXZ(sRiverSoundBgmPos));
```
Might not work because pointer?

**Review by isghj5 — APPROVED** [review]
Kinda crazy how much riversound code is wasting ram buried in a code file instead of the RiverSound actor.

Didn't see anything major

File: src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.h [hunk 4]
```diff
@@ -5,9 +5,44 @@
 
 struct EnRiverSound;
 
+#define RS_GET_TYPE(thisx) ((thisx)->params & 0xFF);
+#define RS_GET_PATH_INDEX(thisx) (((thisx)->params >> 8) & 0xFF);
+
+// Any param not as one of these values will result in UB
```
[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.h:11]: It's listed in the first line of the Wikipedia article, so I'd say so https://en.wikipedia.org/wiki/Undefined_behavior

File: src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c [hunk 5]
```diff
@@ -27,10 +26,68 @@ const ActorInit En_River_Sound_InitVars = {
     (ActorFunc)EnRiverSound_Draw,
 };
 
-#endif
+void EnRiverSound_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c:30]: This pad is definitely a game state cast.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**